content/docs/user-guide/machine-learning-pipelines/defining-pipelines.md

```diff
@@ -0,0 +1,264 @@
+        size: 17397976
+      - path: src/preprocess.py
+        md5: 51627ab6d865c51a634959dbc4914d24
+        size: 14623
+    outs:
+      - path: data/preprocessed
+        md5: 21188b73b5661d4730d769f795462485.dir
+        size: 154683
+        nfiles: 312
+```
+
+</details>
+
+## Stage commands
+
+The command(s) defined in the `stages` (`cmd` field) can be anything your system
+terminal would accept and run directly, for example a shell built-in, an
+expression, or a binary found in `PATH`.
+
+Surround the command with double quotes `"` if it includes special characters
+like `|` or `<`, `>`. Use single quotes `'` instead if there are environment
+variables in it that should be evaluated dynamically.
+
+The same applies to `dvc` helpers -- otherwise they would apply to the DVC call
+itself:
+
+```cli
+$ dvc stage add -n first_stage "./a_script.sh > /dev/null 2>&1"
+$ dvc exp init './another_script.sh $MYENVVAR'
+```
+
+<admon type="warn">
+
+While DVC is platform-agnostic, commands defined in `dvc.yaml` (`cmd` field) may
+only work on some operating systems and require certain software packages or
+libraries in the environment.
+
+</admon>
+
+Commands are executed sequentially until all are finished or until one of them
+fails (see `dvc repro`).
+
+## Simple dependencies
+
+There's more than one type of stage dependency. A simple dependency is a file or
+directory needed for the stage `cmd` to run successfully. When it's contents
+have changed, DVC "invalidates" the stage -- it knows that it needs to run
+again.
+
+<admon type="info">
+
+DVC [calculates a hash] of file/dir contents to compare vs. previous versions.
+This is a distinctive mechanism over traditional build tools like `make`.
+
+[calculates a hash]:
+  /doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory
+
+</admon>
+
+File system-level dependencies are defined in the `deps` list of `dvc.yaml`;
+Alternatively, using the `--deps` (`-d`) option of `dvc stage add` (see the
+previous section's example).
+
+<details>
+
+### External dependencies: click to learn more.
+
+A less common kind of dependency is a _URL dependency_. Instead of files in a
+local disk, you can `dvc import` data from another <abbr>DVC project</abbr> (for
+example hosted on GitHub). External dependencies establish relationships between
+different projects or systems (see `dvc import-url`).
+[Get all the details](/doc/user-guide/external-dependencies).
+
+<admon type="info">
+
+DVC will use special methods to check whether the contents of an URL have
+changed for the purpose of stage invalidation.
+
+</admon>
+
+</details>
+
+## Parameter dependencies
+
+A more ML-specific type of dependency is the _hyperparameter_ (`params` field).
+These are simple values used inside your code to tune data processing, modeling
+attributes, or that determine stage execution in any other way. For example, a
+[random forest classifier] may require a _maximum depth_ value.
```

Comment:
Again, not a strong opinion, but frequently there are also items in `params` that are not ML hyperparameters. It feels a bit out of context to get into these ML-specific details in this guide, especially since there are non-ML users for pipelines.

Reply:
> a bit out of context to get into these ML-specific details in this guide

Good point, you're right (removed ML deets). Should we call the doc "Data Pipelines" though? Using that for now... Just _Pipelines_ is too generic I think.